server/routerlicious/packages/protocol-base/src/quorum.ts

```diff
@@ -111,20 +118,48 @@ export class Quorum extends TypedEventEmitter<IQuorumEvents> implements IQuorum
         this.removeAllListeners();
     }
 
+    /**
+     * Snapshots the entire quorum
+     * @returns a quorum snapshot
+     */
     public snapshot(): IQuorumSnapshot {
+        this.snapshotCache.members ??= this.snapshotMembers();
+        this.snapshotCache.proposals ??= this.snapshotProposals();
+        this.snapshotCache.values ??= this.snapshotValues();
+
+        return {
+            ...this.snapshotCache as IQuorumSnapshot,
+        };
+    }
+
+    /**
+     * Snapshots quorum members
+     * @returns a deep cloned array of members
+     */
+    public snapshotMembers(): IQuorumSnapshot["members"] {
+        return cloneDeep(Array.from(this.members));
+    }
+
+    /**
+     * Snapshots quorum proposals
+     * @returns a deep cloned array of proposals
+     */
+    public snapshotProposals(): IQuorumSnapshot["proposals"] {
         const serializedProposals = Array.from(this.proposals).map(
             ([sequenceNumber, proposal]) => [
                 sequenceNumber,
                 { sequenceNumber, key: proposal.key, value: proposal.value },
                 Array.from(proposal.rejections)] as [number, ISequencedProposal, string[]]);
 
-        const snapshot = {
-            members: [...this.members],
-            proposals: serializedProposals,
-            values: [...this.values],
-        };
+        return cloneDeep(serializedProposals);
```

Comment:
I think you do not need a clone here, as .map() sequence above gives you always new array, no?

Reply:
Yea. It looks like it's already cloned.